File: src/main/java/com/protryon/jasm/stage1/constant/CFloat.java
package com.protryon.jasm.stage1.constant;

import java.io.DataOutputStream;
import java.io.IOException;

import com.protryon.jasm.stage1.CConstant;

public class CFloat extends CConstant<Float> {
	public CFloat(float f) {
		super(f, 4);
	}

	@Override
	public void write(DataOutputStream out) throws IOException {
		out.writeFloat(this.value);
	}
}

File: src/main/java/com/protryon/jasm/stage1/constant/CFloat.kt
package com.protryon.jasm.stage1.constant

import java.io.DataOutputStream
import java.io.IOException

import com.protryon.jasm.stage1.CConstant

class CFloat(f: Float) : CConstant<Float>(f, 4) {

    @Throws(IOException::class)
    public override fun write(out: DataOutputStream) {
        out.writeFloat(this.value)
    }
}
